opennlp-tools/src/main/java/opennlp/tools/util/featuregen/GeneratorFactory.java

```diff
@@ -95,25 +87,235 @@
+
+    final String elementName = element.getTagName();
+
+    // check it is new format?
+    if ("featureGenerators".equals(elementName)) {
+      return addMappingsFromXmlChildren(element.getChildNodes(), new HashMap<>());
+    } else {
+      throw new IllegalArgumentException(
+          "[OPENNLP-1174] - Classic configuration format is no longer supported!");
+    }
+  }
+
+  private static void extractArtifactSerializerMappings(
+      Map<String, ArtifactSerializer<?>> mapping, Element element) {
+    final String className = element.getAttribute("class");
+    if (!className.isBlank()) {
+      try {
+        final Class<?> factoryClass = Class.forName(className);
+        try {
+          final Constructor<?> constructor = factoryClass.getConstructor();
+          final AbstractXmlFeatureGeneratorFactory factory =
+              (AbstractXmlFeatureGeneratorFactory) constructor.newInstance();
+          factory.init(element, null);
+          final Map<String, ArtifactSerializer<?>> map = factory.getArtifactSerializerMapping();
+          if (map != null) {
+            mapping.putAll(map);
+          }
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException
+                 | IllegalAccessException e) {
+          throw new RuntimeException(e);
+        } catch (InvalidFormatException ignored) {
+        }
+      } catch (ClassNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    addMappingsFromXmlChildren(element.getChildNodes(), mapping);
+  }
+
+  private static Map<String, ArtifactSerializer<?>> addMappingsFromXmlChildren(
+      final NodeList nodes, final Map<String, ArtifactSerializer<?>> mapping) {
+    for (int i = 0; i < nodes.getLength(); i++) {
+      if (nodes.item(i) instanceof Element childElem) {
+        if ("generator".equals(childElem.getTagName())) {
+          extractArtifactSerializerMappings(mapping, childElem);
+        }
+      }
+    }
+    return mapping;
+  }
+
+  /**
+   * Provides a list with all the elements in the xml feature descriptor.
+   *
+   * @param xmlDescriptorIn the xml feature descriptor
+   * @return a list containing all elements
+   * @throws IOException            if inputstream cannot be open
+   * @throws InvalidFormatException if xml is not well-formed
+   */
+  public static List<Element> getDescriptorElements(InputStream xmlDescriptorIn)
+      throws IOException {
+
+    final List<Element> elements = new ArrayList<>();
+    final org.w3c.dom.Document xmlDescriptorDOM = createDOM(xmlDescriptorIn);
+    final XPath xPath = XPathFactory.newInstance().newXPath();
+    NodeList allElements;
+    try {
+      final XPathExpression exp = xPath.compile("//*");
+      allElements = (NodeList) exp.evaluate(xmlDescriptorDOM.getDocumentElement(), XPathConstants.NODESET);
+    } catch (XPathExpressionException e) {
+      throw new IllegalStateException("The hard coded XPath expression should always be valid!");
```

Comment:
Please add to JavaDoc that this exception can occur.

Reply:
Should be `InvalidFormatException` here, which is already documented in the JavaDoc but code was misbehaving. Changed to `InvalidFormatException`